CMakeLists.txt

```diff
@@ -69,10 +73,6 @@ IF(NOT "${Boost_VERSION}" MATCHES "1.53(.*)")
    SET(Boost_LIBRARIES ${BOOST_LIBRARIES_TEMP} ${Boost_LIBRARIES})
 ENDIF()
 
-if( NOT CPP_STANDARD )
-  set( CPP_STANDARD "-std=c++11" )
```

Comment:
This PR didn't work for me (Ubuntu 16.04, Gcc 5.4.0, FC revision 8176f39 (latest master)). Changing this line to `c++14` got me further (failed on `wallet.cpp`).

Reply:
Update: after changed `c++11` to `c++14` here, and changed `identity_member` to `identity_member_with_optionals` in wallet.cpp, my build worked. Should I create another PR?

Reply:
Update: I made a mistake, missed the `CMAKE_CXX_STANDARD 14 ` change, I think that's why I failed to build. Now rebuilding.

Reply:
I fired up Ubuntu 16.04.04, this branch, and was able to compile fine.

Reply:
I guess you didn't bump FC.

@pmconrad: how about merging this PR as is, then bump FC in https://github.com/bitshares/bitshares-core/pull/1695?